### PR TITLE
#4398 [Manager] Refactor CreateWorkspace component to use Redux-Form

### PR DESCRIPTION
**Pull request by eechih**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #4398 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/store/epics/workspace/createWorkspaceEpic.js [hunk 1]
```diff
@@ -14,172 +14,301 @@
...
+const getBroker$ = params =>
+  Observable.create(observer => {
+    brokerApi
+      .get(params)
+      .then(res => {
+        observer.next(res);
+        observer.complete();
+      })
+      .catch(error => {
+        observer.error(error);
```
Comment (saivirtue) [inline ohara-manager/client/src/store/epics/workspace/createWorkspaceEpic.js]: 這裡的error會往epic最外層的catchError丟，但我們會在wait的階段不斷重試 ; 是否直接throw exception讓wait來retry就可以了?

Comment (eechih) [reply ohara-manager/client/src/store/epics/workspace/createWorkspaceEpic.js]: getBroker$ 的 error 會先被 waitBroker$ 的 retryWhen 捕獲，不會再往外丟。

Comment (saivirtue) [reply ohara-manager/client/src/store/epics/workspace/createWorkspaceEpic.js]: 了解。那這裡的寫法與wait()中的`throw`是一樣的行為嗎?

Comment (eechih) [reply ohara-manager/client/src/store/epics/workspace/createWorkspaceEpic.js]: > 那這裡的寫法與wait()中的throw是一樣的行為嗎?

我實驗過，是一樣的結果

**Comment by eechih** [comment]
@konekoya @wu87988622 @saivirtue This PR is ready, could you take a review again? thanks.

File: ohara-manager/client/src/components/Node/SelectNodeDialog.js [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+import { NODE_STATE } from 'api/apiInterface/nodeInterface';
+import { KIND, MODE } from 'const';
+import * as hooks from 'hooks';
+import { Dialog } from 'components/common/Dialog';
+import { useConfiguratorState, useViewNodeDialog } from 'context';
+import AddNodeDialog from './AddNodeDialog';
+import ViewNodeDialog from './ViewNodeDialog';
+
+const SelectNodeDialog = React.forwardRef((props, ref) => {
```
Comment (eechih) [inline ohara-manager/client/src/components/Node/SelectNodeDialog.js]: Use Material-Table to rewrite the SelectNodeDialog

File: ohara-manager/client/src/store/epics/workspace/createWorkspaceEpic.js [hunk 3]
```diff
@@ -16,170 +16,177 @@
...
+    catchError(error => of(actions.startWorker.failure(error))),
   );
 
+const finalize$ = params =>
+  of(
+    of(actions.createWorkspace.fulfill()),
+    // Clear form data
+    of(reset(FORM.CREATE_WORKSPACE)),
+    // Back to the first step
+    of(actions.switchCreateWorkspaceStep(0)),
```
Comment (wu87988622) [inline ohara-manager/client/src/store/epics/workspace/createWorkspaceEpic.js]: 為什麼，不幫使用者切換至最新的workspace?

Comment (eechih) [reply ohara-manager/client/src/store/epics/workspace/createWorkspaceEpic.js]: > 為什麼，不幫使用者切換至最新的workspace?

請往下看，有一個 action 負責切換 workspace
```
// Switch to the workspace you just created
of(actions.switchWorkspace.trigger(params)),
```

Comment (wu87988622) [reply ohara-manager/client/src/store/epics/workspace/createWorkspaceEpic.js]: ok~
感謝

File: ohara-manager/client/src/store/epics/workspace/createWorkspaceEpic.js [hunk 4]
```diff
@@ -16,170 +16,177 @@
...
-        ),
-        startWith(actions.createWorkspace.request()),
+    exhaustMap(([action]) => {
+      const { workspace, zookeeper, broker, worker } = action.payload;
+      const workspaceKey = getKey(workspace);
+      const zookeeperKey = getKey(zookeeper);
+      const brokerKey = getKey(broker);
+      const workerKey = getKey(worker);
+      return of(
+        // TODO: upload files
```
Comment (eechih) [inline ohara-manager/client/src/store/epics/workspace/createWorkspaceEpic.js]: will fix on another issue #4459

**Comment by eechih** [comment]
retry

**Comment by eechih** [comment]
retry

**Pull request merged**